src/litserve/specs/openai.py

```diff
@@ -311,35 +306,30 @@ async def options_chat_completions(self, request: Request):
 
     async def chat_completion(self, request: ChatCompletionRequest, background_tasks: BackgroundTasks):
         logger.debug("Received chat completion request %s", request)
-        logger.info("request buffer %s", self._server.request_buffer)
 
         uids = [uuid.uuid4() for _ in range(request.n)]
-        pipes = []
+        self.queues = []
+        self.events = []
         for uid in uids:
-            read, write = self._server.new_pipe()
             request_el = request.model_copy()
             request_el.n = 1
-            self._server.request_buffer[uid] = (request_el, write)
-            self._server.request_queue.put(uid)
-            background_tasks.add_task(self._server.cleanup_request, self._server.request_buffer, uid)
-            pipes.append((read, write))
-
-        responses = await self.get_from_pipes(uids, pipes)
+            q = asyncio.Queue()
```

Comment:
I still think this might be overkill for this specific case. Let's just make a note to ourselves to try a simple `deque`.